src/checker.ml

```diff
@@ -903,15 +904,21 @@ let view_remove_liquidity_min_kit_withdrawn (lqt, state: lqt * checker) : kit =
 
 let view_burrow_max_mintable_kit (burrow_id, state: burrow_id * checker) : kit =
   assert_checker_invariants state;
-  burrow_max_mintable_kit state.parameters (find_burrow state.burrows burrow_id)
+  let burrow = find_burrow state.burrows burrow_id in
+  let burrow = burrow_touch state.parameters burrow in
+  burrow_max_mintable_kit state.parameters burrow
```

Comment:
> Since we're doing this here, maybe we should remove `ensure_uptodate_burrow` call from `burrow_max_mintable_kit` function? The same can apply to other functions too.
> 
> It's not super important since we're on an off-chain view where the gas costs does not matter as much; however the convention of "our functions does not care what kind of a burrow they got, but the entrypoints can assert it or touch the burrow at the beginning" might be good enough.

@utdemir Hmmm, I'd still like some check that those functions are not called on untouched burrows though. You're totally right that `error_OperationOnUntouchedBurrow` is not supposed to be a user-facing error now, but I'd still like the check to happen for my own sanity. I could replace `ensure_uptodate_burrow` with (a) an assertion, or (b) a `failwith` with a string instead of an error code. (I think I prefer (b) since it's not erasable so our e2e tests would catch a regression if I missed something.) What do you think?

Reply:
I'm unsure, when I see a conditional (not an assertion) somewhere I assume that piece of code as "live". An assertion would be my personal preference, so it communicates that it is an invariant rather than a check. But I'm not too concerned and would be happy with any option :).

Reply:
> I'm unsure, when I see a conditional (not an assertion) somewhere I assume that piece of code as "live".

I feel the same way actually so I'll use an assertion :+1: It does make me sad to see it erased though; perhaps we should reconsider our current assertion-erasing policy at some point. Some assertions (like this one) I'd really like to keep.